### ISSUE TITLE
Clean Solution

### ISSUE THREAD

**Issue by ajtruckle** [codecadwallader/codemaid#408]
Hello

I am using Code Maid for the first time. I installed it to sort lines of code.

But I noticed on the menu a "Clean Solution" and I am reluctant to run it! What exactly will it do ? Will is show me the proposals first?

Thank you.

Andrew

**Comment by ralphtrickey** [comment]
I can't answer that, but now would be an excellent time to tell you that you should be using something like Github for source control, that would let you try something like that safely. ;) It's also why I haven't tried it, I'd rather not make my changes more difficult to track with gratuitous changes :(

**Comment by ajtruckle** [comment]
I am using TortoiseGit and Bitbucket.


[😊]


All the same, these features should really be documented rather that you trying really. I think it is a better way forward.


________________________________

**Comment by codecadwallader** [comment]
+1 to the comment on source control, glad you have it. :)

Clean solution will run the same code cleanup that runs on an individual file, but across every available file within the solution.  It will show you a progress dialog where you can cancel between files if desired.  It requires opening each file in the editor so it can be very long running and memory consuming depending on the size of your solution.

Have you checked out http://www.codemaid.net/documentation ?

Hope it helps!

**Comment by ajtruckle** [comment]
I notice in the documentation the screen shots for code cleaning refers to CS files. What about CPP files?

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by codecadwallader** [comment]
Yes, C++ files are cleaned as well.  C# has the highest level of support but we can still do other clean ups (e.g. blank lines inside braces) for other C-style languages.  There's an option to do some very basic cleanups (e.g. eol white space) on any type of file.
